### PR TITLE
feat(scheduler): expand node score range to [0, 100]

### LINKED ISSUE #81281: [Scheduler] Expand the internal range of score to [0, 100]
/sig scheduling
/priority important-soon
/assign

**What would you like to be added**:

We plan to update the internal score range to [0, 100].

**Why is this needed**:

[The [0, 10] range is not differentiating](https://github.com/kubernetes/enhancements/pull/1178#discussion_r311312001) enough. Now that we are transitioning to the framework, we can consider expanding the range.

We'll add two constant into the framework interface and multiply the original score by 10:

https://github.com/kubernetes/kubernetes/blob/92e14d79e91ad49a1fa847cf8d27fc8df84b2cda/pkg/scheduler/framework/v1alpha1/interface.go#L58-L64

https://github.com/kubernetes/kubernetes/blob/92e14d79e91ad49a1fa847cf8d27fc8df84b2cda/pkg/scheduler/core/generic_scheduler.go#L805-L806

This is an internal change and compatible (forward and backwards) the developers won't feel it. 

**See also**:

+ https://github.com/kubernetes/kubernetes/pull/81015 which try to introduce the change.
+ https://github.com/kubernetes/enhancements/pull/1178 

cc/ @bsalamat @ahg-g @Huang-Wei

### PR DESCRIPTION
**Pull request by draveness**
/kind feature
/sig scheduling
/priority important-soon
/assign @ahg-g 

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubernetes/issues/81281

**Special notes for your reviewer**:

It would cause custom priorities broken if they use hand-coded value when calculates the node's score, e.g., 10.

**Does this PR introduce a user-facing change?**:

```release-note
Expand scheduler priority functions and scheduling framework plugins' node score range to [0, 100]. action required. Note: this change is internal and does not affect extender and RequestedToCapacityRatio custom priority, which are still expected to provide a [0, 10] range.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by ahg-g** [comment]
> It would cause custom priorities broken if they use hand-coded value when calculates the node's score, e.g., 10.

arrrrgh, then we have to take care of this before changing the range. The Policy API says that the expected range is [0-10], and so we need to scale that as well similar to what we did for the extender.

So we need to define a value in scheduler/api/types.go, named CustomPriorityMaxScore = 10, and clarify that this is what the [UtilizationShapePoint](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/api/v1/types.go#L141) API expects. The score of the [points](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/factory/plugins.go#L420) should then be scaled to MaxNodeScore.

We also need to change the value for [```defaultFunctionShape```](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/algorithm/priorities/requested_to_capacity_ratio.go#L41) to use MinNodeScore/MaxNodeScore.

**Comment by draveness** [comment]
> > It would cause custom priorities broken if they use hand-coded value when calculates the node's score, e.g., 10.
> 
> arrrrgh, then we have to take care of this before changing the range. The Policy API says that the expected range is [0-10], and so we need to scale that as well similar to what we did for the extender.
> 
> So we need to define a value in scheduler/api/types.go, named CustomPriorityMaxScore = 10, and clarify that this is what the [UtilizationShapePoint](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/api/v1/types.go#L141) API expects. The score of the [points](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/factory/plugins.go#L420) should then be scaled to MaxNodeScore.
> 
> We also need to change the value for [`defaultFunctionShape`](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/algorithm/priorities/requested_to_capacity_ratio.go#L41) to use MinNodeScore/MaxNodeScore.

done

**Review by ahg-g — COMMENTED** [review]
> /kind feature
> /sig scheduling
> /priority important-soon
> /assign @ahg-g
> 
> **What this PR does / why we need it**:
> 
> **Which issue(s) this PR fixes**:
> 
> Fixes #81281
> 
> **Special notes for your reviewer**:
> 
> It would cause custom priorities broken if they use hand-coded value when calculates the node's score, e.g., 10.

This is not the case any more.

> 
> **Does this PR introduce a user-facing change?**:
> 
> ```
> Expand scheduler priority functions and scheduling framework plugins' node score range to [0, 100]. action required.
> ```
> 

Can you also clarify that this change is internal and does not affect extender and  RequestedToCapacityRatio custom priority, which are still expected to provide a [0, 10] range.

> **Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
> 
> ```
> 
> ```

File: pkg/scheduler/factory/plugins.go [hunk 1]
```diff
@@ -417,7 +417,10 @@ func buildScoringFunctionShapeFromRequestedToCapacityRatioArguments(arguments *s
 	n := len(arguments.UtilizationShape)
 	points := make([]priorities.FunctionShapePoint, 0, n)
 	for _, point := range arguments.UtilizationShape {
-		points = append(points, priorities.FunctionShapePoint{Utilization: int64(point.Utilization), Score: int64(point.Score)})
+		points = append(points, priorities.FunctionShapePoint{
+			Utilization: int64(point.Utilization),
+			Score:       int64(point.Score) * (framework.MaxNodeScore / schedulerapi.CustomPriorityMaxScore),
```
[the remark this answers is not in the document]
Comment (draveness) [reply pkg/scheduler/factory/plugins.go]: done

**Comment by ahg-g** [comment]
Thanks Draven, can you please rebase first (there is now taintoleration_test.go, which needs updating), and then squash.

**Comment by draveness** [comment]
> Thanks Draven, can you please rebase first (there is now taintoleration_test.go, which needs updating), and then squash.

done

**Pull request merged**